msg_helpers.go

```diff
@@ -558,6 +558,16 @@ func packDataNsec(bitmap []uint16, msg []byte, off int) (int, error) {
 	if len(bitmap) == 0 {
 		return off, nil
 	}
+	if off > len(msg) {
```

Comment:
I don't think it's possible for us to ever be called with `off > len(msg)`, so I think this check is unneeded.

Reply:
Famous last word :)

The first unit test was testing this and would panic without this test. The cost is marginal so I would tend leave it.

Reply:
It is too. I’m certain we can never hit this condition, but I’ll defer to the unit test.